src/rgw/rgw_lua_background.h

```diff
@@ -11,26 +13,142 @@ namespace rgw::lua {
 constexpr const int INIT_EXECUTE_INTERVAL = 5;
 
 //Writeable meta table named RGW with mutex protection
-using BackgroundMap = std::unordered_map<std::string, std::string>;
-struct RGWTable : StringMapMetaTable<BackgroundMap,
-  StringMapWriteableNewIndex<BackgroundMap>> {
-    static std::string TableName() {return "RGW";}
-    static std::string Name() {return TableName() + "Meta";}
-    static int IndexClosure(lua_State* L) {
-      auto& mtx = *reinterpret_cast<std::mutex*>(lua_touserdata(L, lua_upvalueindex(2)));
-      std::lock_guard l(mtx);
-      return StringMapMetaTable::IndexClosure(L);
+using BackgroundMapValue = std::variant<std::string, int64_t, double, bool>;
+using BackgroundMap  = std::unordered_map<std::string, BackgroundMapValue>;
+
+inline void pushvalue(lua_State* L, const std::string& value) {
+  pushstring(L, value);
+}
+
+inline void pushvalue(lua_State* L, int64_t value) {
+  lua_pushinteger(L, value);
+}
+
+inline void pushvalue(lua_State* L, double value) {
+  lua_pushnumber(L, value);
+}
+
+inline void pushvalue(lua_State* L, bool value) {
+  lua_pushboolean(L, value);
+}
+
+struct RGWTable : EmptyMetaTable {
```

Comment:
in a lot of new code, the pattern is to use the "rgw" namespace, and not append RGW to type names--maybe there's a more descriptive name available

Reply:
it is already part of the `rgw::lua` namespace.
the reason this is called `RGWTable` because it is a lua table exposed as `RGW` to the script